<launch>

    <arg name="cloud_topic" default="/camera/ahat"/>
    <arg name="use_camera" default="false"/>

    <!-- For Processing Rosbag -->
    <param unless="$(arg use_camera)" name="use_sim_time" value="true"/>
    <node unless="$(arg use_camera)" pkg="robot_referencing" type="joint_state_rerouter.py" name="joint_state_rerouter"/> 
    <!-- tfs -->
    <node pkg="multi_aruco" name="aruco_tf_broadcaster" type="aruco_tfs.py"/>

    <node pkg="registration" type="robot_registration" name="registration_node" output="screen">
        <rosparam command="load" file="$(find registration)/config/registration.yaml"/>
        <param name="pcd_dir" value="$(find registration)/pcd_files/"/>
        <remap from="cloud_in" to="$(arg cloud_topic)"/>
    </node> 

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find registration)/rviz/registration.rviz" required="true"/>

    <!-- Robot Render -->
    <include file="$(find abb_irb1200_support)/launch/real_robot_render.launch">
        <!-- arg for using bag file -->
        <arg name="joint_topic" value="/joint_states2"/>
    </include>

</launch>